src/storage/leveldb/store.ts

```diff
@@ -356,11 +362,58 @@ export default class LevelDBStore implements IStore {
       await this.addressIndex.clear();
       await this.walletIndex.cleanWalletData();
     }
+    /* Clean registered nano contracts when cleaning tokens */
+    if (cleanTokens) {
+      await this.nanoContractIndex.clear();
+    }
   }
 
   async cleanMetadata(): Promise<void> {
     await this.tokenIndex.clearMeta();
     await this.addressIndex.clearMeta();
     await this.utxoIndex.clear();
   }
+
+  /**
+   * Return if the nano contract is registered for the given address based on ncKey.
+   *
+   * @param ncKey Pair address:ncId concatenated.
+   * @returns `true` if registered and `false` otherwise.
+   * @async
+   */
+  async isNanoContractRegistered(ncKey: string): Promise<boolean> {
+    return this.nanoContractIndex.isNanoContractRegistered(ncKey);
+  }
+
+  /**
+   * Get a nano contract data on storage from the ncKey.
+   *
+   * @param ncKey Pair address:ncId registered.
+   * @returns Nano contract data instance.
+   * @async
+   */
+  async getNanoContract(ncKey: string): Promise<INcData | null> {
+    return this.nanoContractIndex.getNanoContract(ncKey);
+  }
+
+  /**
+   * Register a nano contract data.
+   *
+   * @param ncKey Pair address:ncId to register as key.
+   * @param ncValue Nano contract basic information.
+   * @async
+   */
+  async registerNanoContract(ncKey: string, ncValue: INcData): Promise<void> {
+    this.nanoContractIndex.registerNanoContract(ncKey, ncValue);
+  }
```

Comment:
All methods here are using `ncKey` in the parameter and docstring, when it should be `ncId`

Reply:
Fixed at https://github.com/HathorNetwork/hathor-wallet-lib/pull/614/commits/f7f228e35c65a12e4ab9eade924898eb8c63348b